mobile/library/common/engine_common.cc

```diff
@@ -5,20 +5,19 @@
 
 namespace Envoy {
 
-// Allows for using the new API listener when building with --copt=-DUSE_API_LISTENER
-#ifdef USE_API_LISTENER
-const char* listener_type = "envoy.listener_manager_impl.api";
-#else
-const char* listener_type = "envoy.listener_manager_impl.default";
-#endif
-
 std::string hotRestartVersion(bool) { return "disabled"; }
 
 EngineCommon::EngineCommon(int argc, const char* const* argv)
-    : options_(argc, argv, &hotRestartVersion, spdlog::level::info, listener_type),
-      base_(options_, real_time_system_, default_listener_hooks_, prod_component_factory_,
-            std::make_unique<PlatformImpl>(), std::make_unique<Random::RandomGeneratorImpl>(),
-            nullptr) {
+    : options_(argc, argv, &hotRestartVersion, spdlog::level::info) {
+  // TODO(alyssar) when this defaults true, move E-M default config over to boostrap config.
+  if (absl::StrContains(options_.configYaml(), "use_api_listener: true")) {
```

Comment:
OK, so we will be able to get rid of that `StrContains` call altogether once we "ship" the light listener manager. I am fine with having it in the meantime.

Reply:
indeed.